### PR TITLE
Storybook: add controls to `Segment` story

### LINKED ISSUE #54957: Add controls to `Segment` story
Add support for [controls](https://storybook.js.org/docs/react/essentials/controls) for the various props in a story.

### PR DESCRIPTION
**Pull request by timagixe**
**What this PR does / why we need it**:
Adds controls to `Segment` story

**Which issue(s) this PR fixes**:

Fixes #54957

**Special notes for your reviewer**:

### CODE REVIEW

**Review by timagixe — COMMENTED** [review]
Hi @JoaoSilvaGrafana, let me know what you think of it.

**Comment by timagixe** [comment]
@JoaoSilvaGrafana, working on `SegmentInput` I have noticed that we pass following props to `input` element which React does not recognize: `allowCustomValue` and `allowEmptyValue`.

Theses produces the following warnings in dev tools.

```
Warning: React does not recognize the `allowCustomValue` prop on a DOM element.
```

```
Warning: React does not recognize the `allowEmptyValue` prop on a DOM element.
```

From my perspective, it looks like the `allowCustomValue` prop should be omitted, since input assumes that the values going to be custom, is not it?

What about the `allowEmptyValue` prop, looks like it should be handled properly or omitted too.

**Comment by Clarity-89** [comment]
> @JoaoSilvaGrafana, working on `SegmentInput` I have noticed that we pass following props to `input` element which React does not recognize: `allowCustomValue` and `allowEmptyValue`.
> 
> Theses produces the following warnings in dev tools.
> 
> ```
> Warning: React does not recognize the `allowCustomValue` prop on a DOM element.
> ```
> 
> ```
> Warning: React does not recognize the `allowEmptyValue` prop on a DOM element.
> ```
> 
> From my perspective, it looks like the `allowCustomValue` prop should be omitted, since input assumes that the values going to be custom, is not it?
> 
> What about the `allowEmptyValue` prop, looks like it should be handled properly or omitted too.

That's a good point, we can just extract these two props during initial destructuring so they don't end up being passed down to the input:
```
export function SegmentInput<T>({
  value: initialValue,
  onChange,
  Component,
  className,
  placeholder,
  inputPlaceholder,
  disabled,
  autofocus = false,
  onExpandedChange,
  // Extracting these two props, so they are not passed down to input
  allowCustomValue,
  allowEmptyValue,
  ...rest
}: React.PropsWithChildren<SegmentInputProps<T>>) {
```

**Comment by timagixe** [comment]
@Clarity-89 

> That's a good point, we can just extract these two props during initial destructuring so they don't end up being passed down to the input:

What do you think of adjusting the type of SegmentInput component with omitting those two props likes this:
```ts
export interface SegmentInputProps<T> extends Omit<SegmentProps<T>, 'allowCustomValue' | 'allowEmptyValue'>, Omit<HTMLProps<HTMLInputElement>, 'value' | 'onChange'> {
  value: string | number;
  onChange: (text: string | number) => void;
}
```

This will help avoiding unused destructured properties.

File: packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx [hunk 1]
```diff
@@ -202,4 +204,59 @@ export const HtmlAttributes = () => {
...
+      if (query) {
+        return result.filter((data) => data.label?.includes(query));
+      }
+      return loadOptions(options);
+    },
+    onChange: ({ value }) => {
+      setValue(value);
+      action('onChange fired')(value);
+    },
+    onExpandedChange: (expanded) => action('onExpandedChange fired')({ expanded }),
```
Comment (timagixe) [inline packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx:227]: Pass `noOptionMessageHandler` handler to props to make it clear in Actions tab when this handler is called.

**Comment by Clarity-89** [comment]
> @Clarity-89
> 
> > That's a good point, we can just extract these two props during initial destructuring so they don't end up being passed down to the input:
> 
> What do you think of adjusting the type of SegmentInput component with omitting those two props likes this:
> 
> ```ts
> export interface SegmentInputProps<T> extends Omit<SegmentProps<T>, 'allowCustomValue' | 'allowEmptyValue'>, Omit<HTMLProps<HTMLInputElement>, 'value' | 'onChange'> {
>   value: string | number;
>   onChange: (text: string | number) => void;
> }
> ```
> 
> This will help avoiding unused destructured properties.

Yeah, I think this could work too.

**Comment by timagixe** [comment]
> Yeah, I think this could work too.

I will create a separate PR for this change to not mess up with storybook changes.

File: packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx (hunk 1 above)
Comment (timagixe) [reply packages/grafana-ui/src/components/Segment/SegmentAsync.story.tsx:227]: @JoaoSilvaGrafana, I have solved this within [bce0928](https://github.com/grafana/grafana/pull/55264/commits/bce0928655f9b326bde5e66f4bb0c24d8e43795b), feel free to review this PR when you have a chance, thank you!

**Review by Clarity-89 — APPROVED** [review]

**Review by JoaoSilvaGrafana — APPROVED** [review]
Great job @timagixe sorry I was on holiday so took a while to reply, thanks @Clarity-89 for reviewing 😄

**Comment by timagixe** [comment]
@JoaoSilvaGrafana, no problem, I hope you enjoyed it! 🚀 

My pleasure to help with this task 🙌

**Pull request merged**